### PR TITLE
Fix the Conda package build

### PR DESCRIPTION
**Pull request by bryant1410**
I saw that [the Conda builds are failing since v4.12](https://github.com/huggingface/transformers/actions/workflows/release-conda.yml). The main problem is that, for some reason, [the build tries to install `setuptools` but Conda build forbids it](https://github.com/huggingface/transformers/runs/5853294684?check_suite_focus=true#step:6:2172). I found [an answer in StackOverflow](https://stackoverflow.com/a/64825075/1165181) that shows it can be fixed by adding the flags ` --single-version-externally-managed --record=record.txt` to the `python setup.py install` command in the `build.sh` file (note the `--record` flag is also needed, otherwise the command fails, stating so).

I also updated the tokenizers version specification, which seemed to have been forgotten to be updated in this file as well.

I added `conda-verify`, which `conda build` uses for some sanity checks.

Finally, I changed `conda-build` to `conda build`, which seems to be the way to use this command.

It'd be good if somebody can check this on their end, to double-check it's working fine:

```bash
conda create -n build-transformers -c huggingface python=3.8 anaconda-client conda-build conda-verify
conda activate build-transformers
TRANSFORMERS_VERSION=$(python setup.py --version) conda build .github/conda
```

### CODE REVIEW

**Comment by bryant1410** [comment]
@sgugger I saw you have worked on the Conda packaging of this repo before. Can you look into it? IMHO this PR doesn't take a lot of time to review.

**Review by sgugger — APPROVED** [review]
LGTM but @LysandreJik is the master on conda :-)

**Comment by bryant1410** [comment]
@LysandreJik friendly ping!

File: .github/conda/meta.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ requirements:
     - sacremoses
     - regex !=2019.12.17
     - protobuf
-    - tokenizers >=0.10.1,<0.11.0
+    - tokenizers >=0.11.1,!=0.11.3,<0.13
```
Comment (LysandreJik) [inline .github/conda/meta.yaml:43]: I wonder if the changes necessary aren't limited to these? It seems like the error originates from `Searching for tokenizers!=0.11.3,<0.13,>=0.11.1`, so I would try it just with this change

**Review by LysandreJik — COMMENTED** [review]
Thank you for your PR! We're likely to move away from having our own conda channel towards maintaining it at the `conda-forge` level, but in the meantime, it will not hurt to merge this PR. Thank you, @bryant1410!

File: .github/conda/meta.yaml (hunk 1 above)
Comment (bryant1410) [reply .github/conda/meta.yaml:43]: I did try with just that but I also got [the other errors that I mentioned in the PR description](https://github.com/huggingface/transformers/runs/5853294684?check_suite_focus=true#step:6:2172).

**Comment by bryant1410** [comment]
@LysandreJik friendly ping :)

**Comment by LysandreJik** [comment]
Sorry, just got to it! I managed to get the build to run correctly by just adding the `tokenizers` line :tada: Let me know if you'd like for me to send logs.

Are you sure we need the rest? I'd be happy to merge your PR with just the tokenizers changes :)

**Comment by bryant1410** [comment]
Yeah. I explained why in the PR description

**Comment by bryant1410** [comment]
> Would you like me to share the logs I got locally showing only the `tokenizers` change was necessary?

I left only the `tokenizers` change now.

**Review by LysandreJik — APPROVED** [review]
Yes, I verified the tests were flaky here. Thanks a lot for your contribution and your patience while we were iterating on it, @bryant1410! Let's get all those conda versions back now.

Thanks, merging :)

**Pull request merged**